### PR TITLE
Introduce and switch to Tests.Helpers, a shared project for test helpers

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
I grew tired of copying code between UT modules.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/2744)

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
Tests do pass locally; remains to be seen how they react in the lab.

File: tests/unittests/UT.common.mk [hunk 1]
```diff
@@ -33,6 +33,10 @@ ifeq ($(strip $(filter EntryPoint.cpp,$(UT_FILES))),)
 _UT_OBJECTS += $(_UT_OBJ_DIR)/EntryPoint.o
 endif
 
+ifneq ($(strip $(filter %.m %.mm,$(UT_FILES))),)
+_UT_OBJECTS += $(patsubst %.mm,$(_UT_OBJ_DIR)/helpers-%.mm.o,$(notdir $(wildcard $(_UT_BASE_DIR)/../helpers/*.mm)))
```
Comment (aballway) [inline tests/unittests/UT.common.mk]: yup this is just as awful as I imagined 😝

File: tests/functionaltests/Tests/NSURLConnectionTests.mm [hunk 2]
```diff
@@ -270,13 +243,13 @@ @implementation NSURLConnectionTestHelper_MultiRunLoop
...
     if (self = [super init]) {
-        _threads = [NSMutableArray<NSThread*> new];
         _loops = [NSMutableArray<NSRunLoop*> new];
+        _spinners = [NSMutableArray<THRunLoopSpinner*> new];
 
         for (size_t i = 0; i < 3; ++i) {
-            NSThread* thread = [[NSThread alloc] initWithTarget:self selector:@selector(spinRunLoop) object:nil];
-            [_threads addObject:thread];
-            [thread start];
+            THRunLoopSpinner* spinner = [THRunLoopSpinner new];
```
Comment (aballway) [inline tests/functionaltests/Tests/NSURLConnectionTests.mm]: does this leak?

Comment (DHowett-MSFT) [inline tests/functionaltests/Tests/NSURLConnectionTests.mm]: nope, ARC!

File: tests/unittests/UT.common.mk (hunk 1 above)
Comment (DHowett-MSFT) [inline tests/unittests/UT.common.mk]: :grin:

Comment (ms-jihua) [inline tests/unittests/UT.common.mk]: uugh~

**Pull request merged**